### PR TITLE
Feature/select key

### PR DESCRIPTION
**Pull request by jeremyletang**

### CODE REVIEW

File: packages/snap/src/transaction.ts [hunk 1]
```diff
@@ -23,9 +41,9 @@ export async function send(node, transaction, sendingMode, publicKey) {
     encodeInputData(transaction, latestBlock),
   ]);
 
-  const keyPair = await deriveKeyPair(0);
+  const keyPair = await findKeyPair(publicKey);
 
-  if (keyPair.publicKey.toString() !== publicKey) {
+  if (keyPair === null) {
     throw invalidParameters('Uknown public key');
```
Comment (emilbayes) [inline packages/snap/src/transaction.ts:47]: Typo in error message

File: packages/snap/src/ui.ts [hunk 2]
```diff
@@ -8,19 +8,23 @@ import { invalidParameters } from './errors';
...
   selectedNetworkEntrypoint: URL,
+  publicKey: string,
 ) {
   const content = panel([
     heading(transactionTitle(transaction)),
     text(`Request from: **${origin}**`),
     divider(),
     ...prettyPrintTx(transaction, text),
     divider(),
+    text(`Selected key: ${publicKey}`),
```
Comment (emilbayes) [inline packages/snap/src/ui.ts:26]: Should this say the public key or the name we gave it? eg. "Snap Key 0"

File: packages/snap/src/transaction.ts (hunk 1 above)
Comment (jeremyletang) [reply packages/snap/src/transaction.ts:47]: oh right, it was there already tho. nice catch

File: packages/snap/src/ui.ts (hunk 2 above)
Comment (jeremyletang) [reply packages/snap/src/ui.ts:26]: I have no opinion about it to be fair.

**Review by emilbayes — APPROVED** [review]

**Pull request merged**